### PR TITLE
Butt slaps fix & Masochism! (fixed)

### PR DESCRIPTION
**Pull request by Sishen1542**
Literally a copy of #8396 but without merge issues due to #8398 
Don't know how to revert a deleted branch (otherwise I would have pushed updates to it) so just close #8396 

Nymphomania previously had no actual trait flag, so when I defined it the file had nothing to pull from.
Whoops!
Ass/face slapping nymphos now keeps tails wagging.
Added a Masochism trait by request, which will allow forced orgasm on ass slaps.
Masochists will also have a chance to gain a LARGE percentage of brute damage on attack with the ability to force orgasm through that as well.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1727,6 +1727,11 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			if(BP)
 				if(damage > 0 ? BP.receive_damage(damage * hit_percent * brutemod * H.physiology.brute_mod, 0) : BP.heal_damage(abs(damage * hit_percent * brutemod * H.physiology.brute_mod), 0))
 					H.update_damage_overlays()
+					if (H.has_trait(TRAIT_MASO))
+						(H.adjustArousalLoss(damage * brutemod * H.physiology.brute_mod))
+					if (H.getArousalLoss() >= 100 && ishuman(H) && H.has_trait(TRAIT_MASO) && H.has_dna())
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
						if (H.getArousalLoss() >= 100 && ishuman(H) && H.has_dna())
```
Both this and the above check revolve around the same trait.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -1727,6 +1727,11 @@ GLOBAL_LIST_EMPTY(roundstart_races)
...
+						(H.mob_climax(forced_climax=TRUE))
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
							H.mob_climax(forced_climax=TRUE)
```

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -1727,6 +1727,11 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			if(BP)
 				if(damage > 0 ? BP.receive_damage(damage * hit_percent * brutemod * H.physiology.brute_mod, 0) : BP.heal_damage(abs(damage * hit_percent * brutemod * H.physiology.brute_mod), 0))
 					H.update_damage_overlays()
+					if (H.has_trait(TRAIT_MASO))
+						(H.adjustArousalLoss(damage * brutemod * H.physiology.brute_mod))
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species.dm]: Also, remove the outer pair of parenthesis here.

File: code/modules/mob/living/carbon/human/species.dm [hunk 4]
```diff
@@ -1727,6 +1727,11 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			if(BP)
 				if(damage > 0 ? BP.receive_damage(damage * hit_percent * brutemod * H.physiology.brute_mod, 0) : BP.heal_damage(abs(damage * hit_percent * brutemod * H.physiology.brute_mod), 0))
 					H.update_damage_overlays()
+					if H.has_trait(TRAIT_MASO)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/species.dm]:
```suggestion
					if(H.has_trait(TRAIT_MASO))
```

**Review by silicons — APPROVED** [review]

**Pull request merged**